cleanlab/multiannotator.py

```diff
@@ -0,0 +1,859 @@
+            lambda s: np.mean(s[pd.notna(s)] == consensus_label_subset[pd.notna(s)]),
+            axis=0,
+        )
+
+    else:
+        raise ValueError(
+            f"""
+            {quality_method} is not a valid annotator quality method!
+            Please choose a valid quality_method: {valid_methods}
+            """
+        )
+
+    return annotator_quality
+
+
+def get_annotator_stats(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.ndarray,
+    consensus_label: np.ndarray,
+    num_annotations: np.ndarray,
+    annotator_agreement: np.ndarray,
+    quality_of_consensus: np.ndarray,
+    quality_method: str = "auto",
+) -> pd.DataFrame:
+    """Returns overall statistics about each annotator.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    pred_probs : np.ndarray
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    consensus_label : np.ndarray
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+    num_annotations : np.ndarray
+        An array of shape ``(N,)`` with the number of annotators that have labeled each example.
+    annotator_agreement : np.ndarray
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    quality_method : str
+        Specifies the method used to calculate the quality of the consensus label.
+        For valid quality methods, view :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`
+
+    Returns
+    -------
+    annotator_stats : pd.DataFrame
+        pandas DataFrame in which each row corresponds to one annotator (the row IDs correspond to annotator IDs), with columns:
+        - ``overall_quality``: overall quality of a given annotator's labels
+        - ``num_labeled``: number of examples annotated by a given annotator
+        - ``agreement_with_consensus``: fraction of examples where a given annotator agrees with the consensus label
+        - ``worst_class``: the class that is most frequently mislabeled by a given annotator
+    """
+
+    def try_get_worst_class(labels, pred_probs):
+        try:
+            return _get_worst_class(labels, pred_probs)
+        except:
+            warnings.warn(
+                "worst_class labels for some annotators are NaN due to missing class labels"
```

Comment:
_get_worst_class function docstring says NaN is returned "if all classes do not have issues" but warning suggests "worst_class labels for some annotators are NaN due to missing class labels". Confusing in documentation which is the real reason for NaN

Reply:
`get_worst_class` function have been edited, above warnings are irrelevant now